contracts/market-wrapper/SuperRareMarketWrapper.sol

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.5;
+
+// ============ External Imports ============
+import {ISuperRareAuctionHouse} from "../external/interfaces/ISuperRareAuctionHouse.sol";
+import { SafeMath } from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+
+// ============ Internal Imports ============
+import {IMarketWrapper} from "./IMarketWrapper.sol";
+
+/**
+ * @title SuperRareMarketWrapper
+ * @author Zach Kolodny
+ * @notice MarketWrapper contract implementing IMarketWrapper interface
+ * according to the logic of SuperRare's NFT Auction House
+ * Original SuperRare NFT AuctionHouse code: https://github.com/pixura/pixura-contracts/blob/master/contracts/src/SuperRareAuctionHouse.sol
+ */
+contract SuperRareMarketWrapper is IMarketWrapper {
+    using SafeMath for uint256;
+
+    // ============ Structs ============
+    struct Token {
+        address contractAddress,
+        uint256 tokenId
+    }
+
+    // ============ Internal Immutables ============
+    ISuperRareAuctionHouse internal immutable auctionHouse;
+
+    // ============ Public Variables ============
+    uint256 public auctionIdTracker;
+    mapping(uint256 => Token) public auctions;
+
+    // ======== Constructor =========
+    constructor(address _superRareAuctionHouse) {
+        auctionHouse = ISuperRareAuctionHouse(_superRareAuctionHouse);
+    }
+
+    // ============ Public Mutable Storage ============
+    uint256 public nextAuctionId = 1;
+
+    // ======== External Functions =========
+
+    /**
+     * @notice SuperRare doesn't use auction Ids so this function 
+     * will always return true
+     * @return TRUE always
+     */
+    function auctionExists(uint256 _auctionId)
+        external
+        view
+        returns (bool)
+    {
+        return true;
+    }
+
+    /**
+     * @notice Determines whether an auction exists/is not finished
+     * since SuperRare doesn't use auctionIds
+     * @return TRUE if the auctionId matches the tokenId + nftContract
+     */
+    function auctionIdMatchesToken(
+        uint256 _auctionId,
+        address _contractAddress,
+        uint256 _tokenId
+    ) external view returns (bool)
+    {
+        ISuperRareAuctionHouse.Auction memory auction = 
+            auctionHouse.getAuctionDetails(_contractAddress, _tokenId);
+
+        auctions[nextAuctionId] = Token(_contractAddress, _tokenId);
+
```

Comment:
did you consider doing a two-way mapping of `Token => auctionId` or `contractAddress => tokenId => auctionId`? this way you can register an auction an infinite number of times, which I don't _think_ is a problem but is a bit odd maybe 🤔